File: MusicPlayer/app/src/main/java/com/example/elina/musicplayer/MusicController.java
package com.example.elina.musicplayer;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import java.util.ArrayList;

public class MusicController implements Callback, SharedPreferences.OnSharedPreferenceChangeListener {
    Button prev;
    Button next;
    Button play;
    Button change;
    TextView nameOfSong;
    TextView artist;
    int position;
    MusicService mediaPlayer;
    ArrayList<Song> songs = MainActivity.getSongList();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_two);

        prev = findViewById(R.id.btn_prev);
        next = findViewById(R.id.btn_next);
        play = findViewById(R.id.btn_play);
        change = findViewById(R.id.btn_settings);
        nameOfSong = findViewById(R.id.tv_title);
        artist = findViewById(R.id.tv_artist);
        position = getIntent().getIntExtra("position", 0);

        prev.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                position--;
                position = position == -1 ? songs.size() - 1 : position;
                mediaPlayer.playPrev();
            }
        });

        next.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                position++;
                position = position == songs.size() ? 0 : position;
                mediaPlayer.playNext();
            }
        });

        play.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mediaPlayer.go();
            }
        });

        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(this);
        String currentTheme = sharedPref.getString("pref_theme","pref_theme_default");
        int themeId = R.style.Theme1;
        switch (currentTheme) {
            case ("Theme1"):
                themeId = R.style.Theme1;
                break;
            case ("Theme2"):
                themeId = R.style.Theme2;
                break;
            case ("Theme3"):
                themeId = R.style.Theme3;
                break;
            default: themeId = R.style.Theme1;
        }
        setTheme(themeId);


        Intent intent = new Intent(this, MusicService.class);
        startService(intent);
        bindService(intent, this, Context.BIND_AUTO_CREATE);

        change.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), SettingsActivity.class);
                startActivity(intent);
            }
        });
    }

    @Override
    public void exactSong(int id) {
        position = id;
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        recreate();
    }
}

File: MusicPlayer/app/src/main/java/com/example/elina/musicplayer/MusicController.kt
package com.example.elina.musicplayer

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.preference.PreferenceManager
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.TextView
import com.example.elina.musicplayer.R.id.*

import java.util.ArrayList

class MusicController : Callback, SharedPreferences.OnSharedPreferenceChangeListener, AppCompatActivity() {
    private lateinit var prev: Button
    private lateinit var next: Button
    private lateinit var play: Button
    private lateinit var change: Button
    private lateinit var nameOfSong: TextView
    private lateinit var artist: TextView
    private var position: Int = 0
    private lateinit var mediaPlayer: MusicService
    private var songs: ArrayList<Song> = MainActivity.getSongList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_two)

        val prev = btn_prev
        val next = btn_next
        val play = btn_play
        val change = btn_settings
        val nameOfSong = tv_title
        val artist = tv_artist
        position = getIntent().getIntExtra("position", 0)

        prev.setOnClickListener {
            position--
            position = if (position == -1) songs.size - 1 else position
            mediaPlayer.playPrev()
        }

        next.setOnClickListener {
            position++
            position = if (position == songs.size) 0 else position
            mediaPlayer.playNext()
        }

        play.setOnClickListener {
            mediaPlayer.go()
        }

        val sharedPref = PreferenceManager.getDefaultSharedPreferences(this)
        val currentTheme = sharedPref.getString("pref_theme", "pref_theme_default")
        var themeId = R.style.Theme1
        when (currentTheme) {
            "Theme1" -> themeId = R.style.Theme1
            "Theme2" -> themeId = R.style.Theme2
            "Theme3" -> themeId = R.style.Theme3
            else -> themeId = R.style.Theme1
        }
        setTheme(themeId)

        val intent = Intent(this, MusicService::class.java)
        startService(intent)
        bindService(intent, this, Context.BIND_AUTO_CREATE)

        change.setOnClickListener {
            val intent = Intent(getApplicationContext(), SettingsActivity::class.java)
            startActivity(intent)
        }
    }

    override fun exactSong(id: Int) {
        position = id
    }

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String) {
        recreate()
    }
}
